### PR TITLE
Final refactor of cloud microphysics parameters

### PR DESCRIPTION
**Pull request by trontrytel**
This PR:
- deletes the big parameter struct
- deletes the CommonTypes module
- moves all the parameters and types into parameters folder
- splits different parameters into smaller structs

### CODE REVIEW

**Comment by trontrytel** [comment]
bors try

**Comment by trontrytel** [comment]
bors try

**Comment by trontrytel** [comment]
bors try

File: src/parameters/AerosolModalNucleation.jl [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+Parameters for pure sulfuric acid nucleation from Dunne et al 1016
+DOI:10.1126/science.aaf2649
 
 # Fields
 $(DocStringExtensions.FIELDS)
 """
-
-Base.@kwdef struct H2S04NucleationParameters{FT} <:
-                   AbstractCloudMicrophysicsParameters
+Base.@kwdef struct H2S04NucleationParameters{FT} <: ParametersType{FT}
```
Comment (sriharshakandala) [inline src/parameters/AerosolModalNucleation.jl:13]: We are using `@kwdef` for structs at many locations. In most cases, default values are not assigned. A normal `struct` would suffice in these cases!

File: docs/src/plots/TerminalVelocityComparisons.jl [hunk 2]
```diff
@@ -170,17 +151,17 @@ function snow_terminal_velocity_individual_Chen(
     return vt
 end
 
-function aspect_ratio_snow_1M(snow::CMT.SnowType, D_r::FT) where {FT <: Real}
-    (; m0, me, Δm, a0, ae, Δa, χm, χa, r0) = snow
-    ρ_i = snow.ρ
-
+function aspect_ratio_snow_1M(snow::CMP.Snow, D_r::FT) where {FT <: Real}
```
Comment (sriharshakandala) [inline docs/src/plots/TerminalVelocityComparisons.jl:154]:
```suggestion
function aspect_ratio_snow_1M(snow::CMP.Snow, D_r::FT) where {FT <: Real}
```

File: src/AerosolActivation.jl [hunk 3]
```diff
@@ -38,7 +33,10 @@ export mean_hygroscopicity_parameter,
 
 Returns a curvature coefficient.
 """
-function coeff_of_curvature(ap::AAP, T::FT) where {FT <: Real}
+function coeff_of_curvature(
+    ap::CMP.AerosolActivationParameters,
+    T::FT,
+) where {FT <: Real}
```
Comment (sriharshakandala) [inline src/AerosolActivation.jl:39]:
```suggestion
) where {FT}
```

File: src/Common.jl [hunk 4]
```diff
@@ -173,7 +171,7 @@ end
 Returns water activity of H2SO4 containing droplet.
 `x` is, for example, 0.1 if droplets are 10 percent sulphuric acid by weight.
 """
-function a_w_xT(H2SO4_prs::APS, tps::TPS, x::FT, T::FT) where {FT <: Real}
+function a_w_xT(H2SO4_prs::HPS, tps::TPS, x::FT, T::FT) where {FT <: Real}
```
Comment (sriharshakandala) [inline src/Common.jl:174]:
```suggestion
function a_w_xT(H2SO4_prs::HPS, tps::TPS, x::FT, T::FT) where {FT}
```

File: src/IceNucleation.jl [hunk 5]
```diff
@@ -91,7 +83,7 @@ Returns the homogeneous freezing nucleation rate coefficient,
 `J`, in m^-3 s^-1 for sulphuric acid solutions.
 Parameterization based on Koop 2000, DOI: 10.1038/35020537.
 """
-function homogeneous_J(ip::APS, Δa_w::FT) where {FT <: Real}
+function homogeneous_J(ip::CMP.Koop2000, Δa_w::FT) where {FT <: Real}
```
Comment (sriharshakandala) [inline src/IceNucleation.jl:86]:
```suggestion
function homogeneous_J(ip::CMP.Koop2000, Δa_w::FT) where {FT}
```

**Comment by sriharshakandala** [comment]
Can we remove prefixed `_` for some variable names?

**Comment by trontrytel** [comment]
> Can we remove prefixed `_` for some variable names?

I tried to find all of them. But in case I missed some we can fix it later, once we spot them.

**Pull request merged**